# London Crime Data

## Introduction

Proyek ini adalah EDA singkat tentang data kejahatan di London dengan Python.

## Reading the Data

Data ini mencakup jumlah laporan kejahatan per bulan, 33 wilayah LSOA, 9 kategori kejahatan besar/32 kejahatan kecil dari Januari 2008-Desember 2016.

In [1]:
import numpy as np
import pandas as pd
import matplotlib as mpl
import matplotlib.pyplot as plt

Mari impor set data London Crime menggunakan metode pandas read_csv().

In [3]:
df_crime = pd.read_csv('E:\Cool Yeah\Hacktiv8\London Crime Data\Data\london_crime_by_lsoa.csv')

Sekarang mari kita lihat 5 baris teratas dari dataset menggunakan fungsi head().

In [4]:
df_crime.head()

lsoa_code     borough               major_category  \
0  E01001116     Croydon                     Burglary   
1  E01001646   Greenwich  Violence Against the Person   
2  E01000677     Bromley  Violence Against the Person   
3  E01003774   Redbridge                     Burglary   
4  E01004563  Wandsworth                      Robbery   

                minor_category  value  year  month  
0  Burglary in Other Buildings      0  2016     11  
1               Other violence      0  2016     11  
2               Other violence      0  2015      5  
3  Burglary in Other Buildings      0  2016      3  
4            Personal Property      0  2008      6

Kita juga bisa melihat 5 baris terbawah dari dataset menggunakan fungsi tail().

In [5]:
df_crime.tail()

lsoa_code     borough               major_category  \
13490599  E01000504       Brent              Criminal Damage   
13490600  E01002504  Hillingdon                      Robbery   
13490601  E01004165      Sutton                     Burglary   
13490602  E01001134     Croydon                      Robbery   
13490603  E01003413      Merton  Violence Against the Person   

                       minor_category  value  year  month  
13490599  Criminal Damage To Dwelling      0  2015      2  
13490600            Personal Property      1  2015      6  
13490601       Burglary in a Dwelling      0  2011      2  
13490602            Business Property      0  2011      5  
13490603                 Wounding/GBH      0  2015      6

Mari kita lihat dimensi dataframe.

In [6]:
df_crime.shape

(13490604, 7)

Kita dapat melihat bahwa data memiliki 13490604 baris dan 7 kolom. Kita akan mengeksplorasi data lebih lanjut.

## Berapakah perubahan jumlah kejadian kejahatan dari tahun 2008 ke tahun 2016?

In [ ]:
num_of_crime = df_crime.loc[df_crime['value'].sum()]

In [12]:
gk = df_crime.groupby(['year', 'major_category'])
gk['value'].sum()

year  major_category             
2008  Burglary                        88092
      Criminal Damage                 91872
      Drugs                           68804
      Fraud or Forgery                 5325
      Other Notifiable Offences       10112
                                      ...  
2016  Other Notifiable Offences       15809
      Robbery                         22528
      Sexual Offences                     0
      Theft and Handling             294133
      Violence Against the Person    232381
Name: value, Length: 81, dtype: int64